src/components/carousel/index.jsx

```diff
@@ -0,0 +1,146 @@
+import { Children, cloneElement, useState } from "react";
+import PropTypes from "prop-types";
+import { useSwipeable } from "react-swipeable";
+
+import Icon from "../icon";
+import Item from "./_children/Item";
+
+const COMPONENT_CLASS_NAME = "c-carousel";
+
+const Carousel = ({ children, className, id, label, next, previous, slidesToShow }) => {
+	const [slide, setSlide] = useState(slidesToShow);
+	const [position, setPosition] = useState(0);
+	const containerClassNames = [COMPONENT_CLASS_NAME, className].filter((i) => i).join(" ");
+
+	const subComponents = Object.keys(Carousel).map((key) =>
+		Children.map(children, (child) => (child?.type?.name === key ? child : null))
+	);
+
+	const childItems = Children.toArray(subComponents);
+
+	const carouselItems = childItems.map((child, index) =>
+		child.type.name === "Item"
+			? cloneElement(child, {
+				slideNumber: index + 1,
+				viewable: index + 1 > slide - slidesToShow && index + 1 <= slide,
+			  })
+			: null
+	);
+
+	const previousSlide = () => {
+		if (slide - 1 < slidesToShow) {
+			return;
+		}
+		setSlide(slide - 1);
+		setPosition(position + 100 / slidesToShow);
+	};
+
+	const nextSlide = () => {
+		if (slide + 1 > carouselItems.length) {
+			return;
+		}
+		setSlide(slide + 1);
+		setPosition(position - 100 / slidesToShow);
+	};
+
+	/* istanbul ignore next */
+	const handlers = useSwipeable({
+		onSwipedLeft: () => nextSlide(),
+		onSwipedRight: () => previousSlide(),
+		preventDefaultTouchmoveEvent: true,
+		trackMouse: true,
+	});
+
+	return (
+		<div
+			className={containerClassNames}
+			id={id}
+			aria-label={label}
+			role="region"
+			aria-roledescription="carousel"
+			style={{ "--carousel-slide-width": slidesToShow !== 4 ? `${100 / slidesToShow}%` : null }}
+		>
+			<div
+				className="c-carousel__track"
+				style={{ transform: `translate3d(${position}%, 0px, 0px)` }}
+				{...handlers}
```

Comment:
not sure I see the utility here with the handlers?

Reply:
The `handlers` is set on line 47 - making use of a library to give us a consistent interface to swipe events rather than having to implement completely custom touch events

Reply:
locally on phone still not reliably scrolling. maybe it's just me

https://user-images.githubusercontent.com/5950956/161303721-67b7f4e3-9ae8-4962-ab8c-b6f3d8b0048d.MP4

Reply:
@JackHowa - it's hard to tell what the issue you are seeing is, as I see the page moving up and down which indicates you trying to scroll vertically - which isn't something that will trigger on the carousel. Then I see the carousel move which I assume is because you swiped left

Reply:
I think `.c-carousel__actions` is getting in the way of the touch events.  I can get it to fairly reliably work if I add `trackTouch: true` to the handlers and swipe either the top or bottom of the carousel.

Reply:
I've moved the handles to the parent element and I get a more consistent experience now

Reply:
added what fixed touch events for me (tested on my android phone)

Reply:
@JackHowa : See if the issue still exists if you can.  I am unable to reproduce issues after the latest update.